parsons/scytl/scytl.py

```diff
@@ -0,0 +1,561 @@
+import urllib
+import zipfile
+import csv
+import requests
+import urllib.error
+import urllib.request
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from dateutil.parser import parse as parsedate
+from pytz import timezone
+from io import BytesIO, StringIO
+
+clarity_url = 'https://results.enr.clarityelections.com/'
+
+get_version_url = \
+    clarity_url + '{administrator}/{election_id}/current_ver.txt'
+summary_csv_url_template = \
+    clarity_url + '{administrator}//{election_id}/{version_num}/reports/summary.zip'
+detail_xml_url_template = \
+    clarity_url + '{administrator}/{election_id}/{version_num}/reports/detailxml.zip'
+county_detail_xml_url = clarity_url + \
+    '{state}/{county_name}/{county_election_id}/{county_version_num}/reports/detailxml.zip'
+detail_settings_json_url_template = \
+    clarity_url + '{state}/{election_id}/{version_num}/json/en/electionsettings.json'
+
+tzinfos = {
+    "EST": "UTC-5",
+    "EDT": "UTC-4",
+    "CST": "UTC-6",
+    "CDT": "UTC-5",
+    "MST": "UTC-7",
+    "MDT": "UTC-6",
+    "PST": "UTC-8",
+    "PDT": "UTC-7",
+    "AKST": "UTC-9",
+    "AKDT": "UTC-8",
+    "HST": "UTC-10",
+    "HDT": "UTC-9",
+}
+
+
+class CountyDetails:
+    def __init__(
+        self,
+        state: str,
+        county_name: str,
+        county_election_id: str,
+        county_version_num: str,
+        county_update_date: datetime = None
+    ):
+        self.state = state
+        self.county_name = county_name
+        self.county_election_id = county_election_id
+        self.county_version_num = county_version_num
+        self.county_update_date = county_update_date
+
+
+class Scytl(object):
+    """
+    Instantiate a Scytl connector.
+
+    `Args:`:
+        state: str
+            The two letter code of the state the publishing election results.
+            ex: GA
+        election_id: str
+            The numeric identifier for the election found in the url of the election's website.
+            ex: "114729"
+        county: str (optional)
+            The name of the county publishing the results.
+            ex: Clarke
+    """
+
+    def __init__(self, state: str, election_id: str, county=""):
+        self.state = state
+        self.county = county.replace(' ', '_')
+
+        self.administrator = f"{self.state}/{self.county}" if self.county else self.state
+        self.election_id = election_id
+
+        self.previous_summary_version_num = None
+        self.previous_details_version_num = None
+        self.previous_county_details_version_num = None
+        self.previous_county_details_list = None
+        self.previously_fetched_counties = set([])
+
+    def _parse_date_to_utc(self, input_dt):
```

Comment:
Can you document the private methods? This is not for the users, since they won't display in our docs, but very helpful for future maintainability/readability.

Reply:
Updated!